pkg/discovery/repository/kube_cluster.go

```diff
@@ -33,76 +34,144 @@ type KubeCluster struct {
 
 	K8sAppToComponentMap map[K8sApp][]K8sAppComponent
 	ComponentToAppMap    map[K8sAppComponent][]K8sApp
+	ControllerMap        map[string]*K8sController
 }
 
-type K8sApp struct {
-	Uid       string
-	Namespace string
-	Name      string
+func NewKubeCluster(clusterName string, nodes []*v1.Node) *KubeCluster {
+	return &KubeCluster{
+		Name:         clusterName,
+		Nodes:        nodes,
+		Pods:         []*v1.Pod{},
+		NamespaceMap: make(map[string]*KubeNamespace),
+	}
 }
 
-type K8sAppComponent struct {
-	EntityType proto.EntityDTO_EntityType
-	Uid        string
-	Namespace  string
-	Name       string
+func (kc *KubeCluster) WithPods(pods []*v1.Pod) *KubeCluster {
+	kc.Pods = pods
+	return kc
 }
 
-type PodVolume struct {
-	// Namespace qualified pod name.
-	QualifiedPodName string
-	// Name used by the pod to mount the volume.
-	MountName string
+// Summary object to get the nodes, quotas and namespaces in the cluster
+type ClusterSummary struct {
+	*KubeCluster
+	// Computed
+	NodeMap                  map[string]*KubeNode
+	NodeNameUIDMap           map[string]string
+	NamespaceUIDMap          map[string]string
+	PodClusterIDToServiceMap map[string]*v1.Service
+	NodeToRunningPods        map[string][]*v1.Pod
+	NodeToPendingPods        map[string][]*v1.Pod
```

Comment:
Pending pods are scheduled yet, right?  What Node does this map refer to?

Reply:
The Pending pods here are already scheduled (bound) to a  node, but they may still be pulling images, or stuck in `ImagePullBackOff`.

The `kube-system/coredns-7c4d89dd66-jr5zj` pod in CCP1 cluster is an example.